src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/EnumConverter.cs

```diff
@@ -39,6 +41,14 @@ internal sealed class EnumConverter<T> : JsonPrimitiveConverter<T>
         // This is used to prevent flooding the cache due to exponential bitwise combinations of flags.
         // Since multiple threads can add to the cache, a few more values might be added.
         private const int NameCacheSizeSoftLimit = 64;
+        private const string NumericPattern = @"^\s*(\+|\-)?[0-9]+\s*$";
+
+#if NETCOREAPP
+        [GeneratedRegex(NumericPattern)]
+        private static partial Regex NumericRegex();
+#else
+        private static readonly Regex s_numericRegex = new(NumericPattern);
```

Comment:
Can't we also use the source generator to target non .NET core TFMs?

Reply:
@lateapexearlyspeed that would basically involve including the source code for `GeneratedRegexAttribute.cs` in older targets and then use `[GeneratedRegex(NumericPattern)]` everywhere. I expect that this would cause the referenced source generator to kick in.

Reply:
After including `GeneratedRegexAttribute.cs` (and its dependeny file) in current project, non-coreapp targets can recognize `GeneratedRegexAttribute` but compilation will fail about code-gen as follow:
![image](https://github.com/dotnet/runtime/assets/72254037/5b6ef05d-df25-4532-8f32-150a1e50ba47)

Reply:
I'm not surprised about that. I don't think that the Regex source gen is designed to work on older frameworks. cc @stephentoub

Reply:
Fair, that was just me naively assuming it wouldn't require new APIs.